### PR TITLE
Use py::class_ for rcl_duration_t

### PR DESCRIPTION
**Pull request by ahcorde**
Part of #665

This is a PR that refactors the code using `rcl_duration_t` to use a `py::class_` definition. 

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: rclpy/src/rclpy/qos.cpp [hunk 1]
```diff
@@ -91,10 +86,10 @@ convert_from_py_qos_policy(
   int qos_depth,
   int qos_reliability,
   int qos_durability,
-  py::capsule pyqos_lifespan,
-  py::capsule pyqos_deadline,
+  rcl_duration_t pyqos_lifespan,
+  rcl_duration_t pyqos_deadline,
   int qos_liveliness,
-  py::capsule pyqos_liveliness_lease_duration,
+  rcl_duration_t pyqos_liveliness_lease_duration,
```
Comment (hidmic) [inline rclpy/src/rclpy/qos.cpp:92]: @ahcorde nit: could be made const ref.

File: rclpy/src/rclpy/qos.cpp [hunk 2]
```diff
@@ -91,10 +86,10 @@ convert_from_py_qos_policy(
   int qos_depth,
   int qos_reliability,
   int qos_durability,
-  py::capsule pyqos_lifespan,
-  py::capsule pyqos_deadline,
+  rcl_duration_t pyqos_lifespan,
+  rcl_duration_t pyqos_deadline,
```
Comment (hidmic) [inline rclpy/src/rclpy/qos.cpp:90]: @ahcorde nit: these two could be made const ref.

File: rclpy/src/rclpy/qos.cpp [hunk 3]
```diff
@@ -72,15 +72,10 @@ namespace
 
 // Fill a given rmw_time_t from an rclpy.duration.Duration instance.
 void
-_convert_py_duration_to_rmw_time(py::capsule pyduration, rmw_time_t * out_time)
+_convert_py_duration_to_rmw_time(rcl_duration_t duration, rmw_time_t * out_time)
```
Comment (hidmic) [inline rclpy/src/rclpy/qos.cpp:75]: @ahcorde nit: could be made const ref.

**Review by hidmic — APPROVED** [review]
LGTM pending green CI

**Comment by ahcorde** [comment]
build up-to `rclpy` and testing `rclpy`

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14165/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8964/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11845/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14264/)

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14166/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8965/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11846/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14265/)

**Comment by ahcorde** [comment]
The warning on Windows is unrelated.

**Pull request merged**